### PR TITLE
Hicexplorer hicpca

### PR DESCRIPTION
**Pull request by jianhong**
## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/hicexplorer/hicpca/main.nf [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        'quay.io/biocontainers/hicexplorer:3.7.2--pyhdfd78af_1' }"
+
+    input:
+    tuple val(meta), path(matrix)
+
+    output:
+    tuple val(meta), path("${prefix}_*")              , emit:results
+    tuple val(meta), path("${prefix}_pca1.$format")   , emit:pca1
+    tuple val(meta), path("${prefix}_pca2.$format")   , emit:pca2
+    path("versions.yml")                              , emit:versions
```
Comment (sguizard) [inline modules/nf-core/hicexplorer/hicpca/main.nf:17]:
```suggestion
    tuple val(meta), path("${prefix}_*")           , emit:results
    tuple val(meta), path("${prefix}_pca1.$format"), emit:pca1
    tuple val(meta), path("${prefix}_pca2.$format"), emit:pca2
    path("versions.yml")                           , emit:versions
```

File: modules/nf-core/hicexplorer/hicpca/main.nf [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
+    args = args.tokenize()
+    def idx = args.findIndexOf{ it == '--format' | it == '-f' }
+    format = 'bigwig'
```
Comment (sguizard) [inline modules/nf-core/hicexplorer/hicpca/main.nf:27]:
```suggestion
    def args = task.ext.args ?: ''
    prefix   = task.ext.prefix ?: "${meta.id}"
    args     = args.tokenize()
    def idx  = args.findIndexOf{ it == '--format' | it == '-f' }
    format   = 'bigwig'
```

File: modules/nf-core/hicexplorer/hicpca/meta.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+name: "hicexplorer_hicpca"
+description: Computes PCA eigenvectors for a Hi-C matrix.
+keywords:
+  - sort
```
Comment (sguizard) [inline modules/nf-core/hicexplorer/hicpca/meta.yml:4]: More keywords needed please.

**Review by sguizard — CHANGES_REQUESTED** [review]
LGTM. 
Just added 2 suggestions and asked to add more keywords to meta file.

**Comment by jianhong** [comment]
@sguizard Could you please take a second view?

**Review by sguizard — APPROVED** [review]

**Pull request merged**